File: 27. Remove Element/Solution.py
class Solution:
    def removeElement(self, nums, val):
        """
        type nums: List[int]
        type val: int
        rtype: int
        """
        l, r = 0, len(nums)
        while (l < r):
            if nums[l] == val:
                nums[l] = nums[r - 1]
                r -= 1
            else:
                l += 1
        return r

File: 27. Remove Element/Solution.cpp
#include <vector>

using std::vector;

class Solution {
public:
    int removeElement_1(vector<int>& nums, int val) {
        int l = 0, r = nums.size();
        while(l < r)
        {
            if (nums[l] == val)
            {
                nums[l] = nums[r - 1];
                r--;
            }
            else
                l++;
        }
        return r;
    }
};